### PR TITLE
kompanion: iterate over kinds by default

### PR DESCRIPTION
**Pull request by justinsb**

### CODE REVIEW

File: experiments/kompanion/cmd/export/export.go [hunk 1]
```diff
@@ -280,25 +304,46 @@ func RunExport(ctx context.Context, opts *ExportOptions) error {
 		return fmt.Errorf("could not create %q directory: %w", reportTempDir, err)
 	}
 
+	shouldExcludeObject := func(id types.NamespacedName) bool {
+		if shouldExclude(id.Namespace, opts.ignoreNamespaces, opts.targetNamespaces) {
+			return true
+		}
+		return shouldExclude(id.Name, opts.ignoreObjects, opts.targetObjects)
+	}
```
Comment (yuwenma) [inline experiments/kompanion/cmd/export/export.go:312]: The logic is kind of twisted. I'm wondering if we can check GKNN (as a whole struct)  instead.

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve 

One minor regarding the complexity of the `shouldExclude` logic.

File: experiments/kompanion/cmd/export/export.go [hunk 2]
```diff
@@ -280,25 +304,46 @@ func RunExport(ctx context.Context, opts *ExportOptions) error {
...
+				Resources:           []schema.GroupVersionResource{resource},
+				DynamicClient:       dynamicClient,
+				ReportDir:           reportTempDir,
+				ShouldExcludeObject: shouldExcludeObject,
+			})
+		}
 	}
 
-	log.Printf("Starting worker threads to process namespaces")
+	log.Printf("Starting worker threads to process KCC objects")
```
Comment (yuwenma) [inline experiments/kompanion/cmd/export/export.go:346]: super nit: KCC --> Config Connector

File: experiments/kompanion/cmd/export/export.go (hunk 1 above)
Comment (justinsb) [reply experiments/kompanion/cmd/export/export.go:312]: Yes, the challenge is that we also want to check the namespace individually.  But yes, I think we can move in this direction!

**Pull request merged**